crates/uv-python/src/installation.rs

```diff
@@ -88,7 +88,7 @@ impl PythonInstallation {
         // Perform a fetch aggressively if managed Python is preferred
         if matches!(preference, PythonPreference::Managed) && python_fetch.is_automatic() {
             if let Some(request) = PythonDownloadRequest::try_from_request(&request) {
-                return Self::fetch(request, client_builder, cache).await;
+                return Self::fetch(request.fill(), client_builder, cache).await;
```

Comment:
Alternatively, perhaps we should be doing this a bit further down in `Self::fetch`?

Reply:
Or in `ManagedPythonDownload::from_request` to make sure we never miss it?

Reply:
Why don't we have to do it on line 106?

Reply:
We should. Great indication that we should enforce this elsewhere haha

Reply:
Lol

Reply:
Could they have distinct types?

Reply:
Like `fill` returns a different type that you can only get by calling `fill`. That would have the benefit of making it impossible to miss while also giving us the flexibility to `fill` when we want (vs. coupling with the constructor).

Reply:
Yes that sounds reasonable but there are a weird amount of requests types to cast between already and I'd want to think a little more holistically if we add another.